### DISCUSSION TITLE
Adding autoscaling to CosmosDB containers.

### DISCUSSION THREAD

**Discussion by deniz-newday** [pulumi/pulumi#11098] [General]
Hi,

I have a CosmosDb (Sql) account with Provisioned Throughput turned on. At the moment, all containers have the default 400 RU. I want to set autoscaling on some of the containers as their usage varies. But I get the below error
`* Code="BadRequest" Message="Updating offer to autoscale throughput is not allowed. Please invoke migration API to migrate this offer.\r\nActivityId: ead8f411-3e2d-4aec-836b-63623b28219c, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0"`

Here is the code that I use to create the database and the container. 
```
var databaseAccount = new DatabaseAccount("name", new DatabaseAccountArgs
        {
            ResourceGroupName = "ResourceGroupName",
            DatabaseAccountOfferType = DatabaseAccountOfferType.Standard,
            ConsistencyPolicy = new ConsistencyPolicyArgs
            {
                DefaultConsistencyLevel = DefaultConsistencyLevel.Strong
            },
            EnableAutomaticFailover = true,
            Locations = locations,
            Capabilities = new InputList<CapabilityArgs>(),
            Tags = _tags
        }, options);

var sqlResource = new SqlResourceSqlDatabase(DatabaseName, new SqlResourceSqlDatabaseArgs
        {
            ResourceGroupName = "ResourceGroupName",
            AccountName = DatabaseAccount.Name,
            Resource = new SqlDatabaseResourceArgs
            {
                Id = DatabaseName
            },
            Tags = _tags
        });

var indexes = new List<IndexPathArgs>();
var container = new SqlResourceSqlContainer("containerName", new SqlResourceSqlContainerArgs
      {
          AccountName = databaseAccount.Name,
          ContainerName = "containerName",
          DatabaseName = sqlResource .Name,
          Location = "location",
          Resource = new SqlContainerResourceArgs
          {
              Id = "containerName",
              PartitionKey = new ContainerPartitionKeyArgs
              {
                  Kind = "Hash",
                  Paths =
                  {
                      "partitionKey"
                  }
              },
              IndexingPolicy = new IndexingPolicyArgs
              {
                  Automatic = true,
                  IndexingMode = "Consistent",
                  IncludedPaths = indexes
                      .Select(p => new IncludedPathArgs {Path = p.ToPath()}).ToList(),
                  ExcludedPaths = new List<ExcludedPathArgs>
                  {
                      new()
                      {
                          Path = "/*"
                      }
                  }
              }
          },
          Options = new CreateUpdateOptionsArgs()
          {
              AutoscaleSettings = new AutoscaleSettingsArgs() { MaxThroughput = maxThroughput.Value }
          },
          ResourceGroupName = "resourceGroupName",
          Tags = tags,
      });
```

Does anyone have done a similar thing to update existing CosmosDb container to autoscaling?

**Comment by pierskarsenbarg** [comment]
Hi Deniz,

According to the [Azure documentation](https://learn.microsoft.com/en-us/azure/cosmos-db/autoscale-faq#can-i-enable-autoscale-on-an-existing-database-or-a-container-), it's not possible to enable autoscaling on an existing database or a container using ARM templates. This means you will not be able to do it with the Pulumi Azure Native provider. Usually I'd suggest calling one of the Azure APIs using the SDK, but in this case it doesn't look like you can do it there either.

It does look like you can do it using the Azure CLI, so you could use the [Pulumi command provider](https://www.pulumi.com/registry/packages/command) and run the command to enable it using that.
